File: Android/Soccer_442/app/src/main/java/com/oms/lindanyoka/soccer_442/AppConfig.java
package com.oms.lindanyoka.soccer_442;

/**
 * Created by linda.nyoka on 2015-02-22.
 */
public class AppConfig {

    static String BPL = "barclays-premier-league";
    static String FA = "fa-cup";
    static String AFCON = "africa-cup-of-nations";
    static String LaLiga = "spain";
    static String Bundasliga = "germany";
    static String Absa = "absa-premiership";
    static String League1 = "france";
    static String serieA = "italy";
    static String UEFA = "uefa-champions-league";
    static String UEFA_Euro = "uefa-euro";
    static String copa = "copa-del-rey";

    public static String baseUri = "mobi.supersport.com";
    public static String LogUri = "mobi.supersport.com/football/%s/logs";
    public static String MatchDetails = "mobi.supersport.com/football/%s/match/";//"{1}/commentary";
    public static String MatchDetailsPast= "mobi.supersport.com/football/match/%s/commentary";
    public static String LiveUri = "mobi.supersport.com/football/%s";
    public static String FixtureUri = "www.supersport.com/football/%s/fixtures";
    public static String ResultUri = "www.supersport.com/football/%s/Results";
    public static String ResultUri2 = "mobi.supersport.com/football/%s/Results";
    public static String ScorersUri = "www.supersport.com/football/%s/scorers";
    public static String StatsUri = "mobi.supersport.com/football/match/%s/stats";
    public static String LineUpUri = "mobi.supersport.com/football/match/%s/lineup";
    public static String News = "mobi.supersport.com/football/%s/news";

    public String GetMatchDetails(Competition cometition, long matchId)
    {
        return GetUri(MatchDetails, cometition) + String.valueOf(matchId) + "/commentary";
    }
    public String GetFixtureUri(Competition competition)
    {
        return GetUri(FixtureUri, competition);
    }
    public String GetResultsUri(Competition competition)
    {
        return GetUri(ResultUri, competition);
    }
    public String GetResultsUri2(Competition competition)
    {
        return GetUri(ResultUri2, competition);
    }
    public String News(Competition competition)
    {
        return GetUri(News, competition);
    }
    public String NewsArticle(String uri)
    {
        return baseUri + uri;
    }
    public String GetStatsUri(String matchId)
    {
        return String.format(StatsUri,matchId);
        //return GetUri(ResultUri2, competition);
    }
    public String GetLineUpUri(String matchId)
    {
        return String.format(LineUpUri,matchId);
        //return GetUri(ResultUri2, competition);
    }
    public String GetScorersUr(Competition competition)
    {
        return GetUri(ScorersUri, competition);
    }
    public String GetLogUri(Competition competition)
    {
        return GetUri(LogUri, competition);
    }
    public String GetLiveUri(Competition competition)
    {
        return GetUri(LiveUri, competition);
    }
    private String GetUri(String uri, Competition competition)
    {
        switch (competition)
        {
            case Absa: return String.format(uri, Absa);
            //case AFCON: return String.format(uri, AFCON);
            case BPL: return String.format(uri, BPL);
            case Bundesliga: return String.format(uri, Bundasliga);
            case Copa: return String.format(uri, copa);
            case FA: return String.format(uri, FA);
            case LaLiga: return String.format(uri, LaLiga);
            case League1: return String.format(uri, League1);
            case SerieA: return String.format(uri, serieA);
            case UEFA: return String.format(uri, UEFA);
            //case UEFA_Euro: return String.format(uri, UEFA_Euro);
        }
        return null;
    }
}

File: Windows phone/442/442/442/Api/AppConfig.cs
using System.Globalization;

namespace _442.Api
{
    public class AppConfig
    {

        #region Properties

        const string BPL = "barclays-premier-league";
        const string FA = "fa-cup";
        const string AFCON = "africa-cup-of-nations";
        const string LaLiga = "spain";
        const string Bundasliga = "germany";
        const string Absa = "absa-premiership";
        const string League1 = "france";
        const string serieA = "italy";
        const string UEFA = "uefa-champions-league";
        const string UEFA_Euro = "uefa-euro";
        const string copa = "copa-del-rey";

        public const string LogUri = "mobi.supersport.com/football/{0}/logs";
        public const string MatchDetails = "mobi.supersport.com/football/{0}/match/";//"{1}/commentary";
        public const string LiveUri = "mobi.supersport.com/football/{0}";
        public const string FixtureUri = "www.supersport.com/football/{0}/fixtures";
        public const string ResultUri = "www.supersport.com/football/{0}/Results";
        public const string ScorersUri = "www.supersport.com/football/{0}/scorers";
        #endregion Properties

        #region Methods
        public string GetMatchDetails(Competition cometition, long matchId)
        {
            return GetUri(MatchDetails, cometition) + matchId.ToString
                (CultureInfo.InvariantCulture) + "/commentary";
        }
        public string GetFixtureUri(Competition competition)
        {
            return GetUri(FixtureUri, competition);
        }
        public string GetResultsUri(Competition competition)
        {
            return GetUri(ResultUri, competition);
        }
        public string GetScorersUr(Competition competition)
        {
            return GetUri(ScorersUri, competition);
        }
        public string GetLogUri(Competition competition)
        {
            return GetUri(LogUri, competition);
        }
        public string GetLiveUri(Competition competition)
        {
            return GetUri(LiveUri, competition);
        }
        private string GetUri(string uri, Competition competition)
        {
            switch (competition)
            {
                case Competition.Absa: return string.Format(uri, Absa);
                case Competition.AFCON: return string.Format(uri, AFCON);
                case Competition.BPL: return string.Format(uri, BPL);
                case Competition.Bundesliga: return string.Format(uri, Bundasliga);
                case Competition.Copa: return string.Format(uri, copa);
                case Competition.FA: return string.Format(uri, FA);
                case Competition.LaLiga: return string.Format(uri, LaLiga);
                case Competition.League1: return string.Format(uri, League1);
                case Competition.SerieA: return string.Format(uri, serieA);
                case Competition.UEFA: return string.Format(uri, UEFA);
                case Competition.UEFA_Euro: return string.Format(uri, UEFA_Euro);
            }
            return null;
        }

        #endregion Methods
    }
}
